packages/mui-styled-engine-sc/src/index.d.ts

```diff
@@ -312,3 +325,22 @@ export type CreateMUIStyled<
   MuiStyledOptions extends object = {},
   T extends object = {},
 > = ThemedBaseStyledInterface<MUIStyledCommonProps, MuiStyledOptions, AnyIfEmpty<T>>;
+
+export type PropsOf<C extends keyof JSX.IntrinsicElements | React.JSXElementConstructor<any>> =
+  JSX.LibraryManagedAttributes<C, React.ComponentProps<C>>;
+
+export interface MUIStyledComponent<
```

Comment:
This is the most important change. `StyledComponent` from emotion & styled-components had different signatures, so I had to create a common `MUIStyledComponent` that we can use in the system package.